### PR TITLE
feat: add support for keyboard initialized menu popup

### PR DESCRIPTION
**Pull request by MarshallOfSound**
Solves some accessibility issues with `menu.popup` for keyboard nav accessibility users

This PR also floats a patch that I intend to upstream.  Chrome does not follow standard accessibility behaviour currently (when compared to native menus and even the //views implementation of menus that Edge uses) so this patch should go upstream to benefit all screen reader users too 👍 

Notes: Added support for flagging a Menu as being keyboard initiated

### CODE REVIEW

File: docs/api/menu.md
Comment (erickzhao) [inline docs/api/menu.md:86]: docs: I didn't really like the "certain focus behaviors" wording so here's an attempt at being more precise.

```suggestion
  * `initiatedByKeyboard` boolean (optional) _Windows_ _Linux_ - Whether the menu popup
    was triggered by a keyboard action. If true, the first top-level item in the menu
    will be focused when the menu is opened. This behavior can assist in keyboard
    accessibility.
```

Comment (erickzhao) [reply docs/api/menu.md:86]: api: Do we want to make this API something like `focusFirstItem` instead?

It seems a bit counter-intuitive that "Initiated By Keyboard" is exposed as an API option for the developer to wire up in their JavaScript code rather than something that's auto-detected.

Also, do we ever foresee using any of the other [MenuSourceType](https://source.chromium.org/chromium/chromium/src/+/main:ui/base/ui_base_types.h;bpv=1;bpt=1;l=133?q=MENU_SOURCE_KEYBOARD&ss=chromium&gsn=MenuSourceType&gs=KYTHE%3A%2F%2Fkythe%3A%2F%2Fchromium.googlesource.com%2Fcodesearch%2Fchromium%2Fsrc%2F%2Fmain%3Flang%3Dc%252B%252B%3Fpath%3Dui%2Fbase%2Fui_base_types.h%23MenuSourceType%253Aui%2523n%2523TX9FKDk8TS) options available?

Comment (MarshallOfSound) [reply docs/api/menu.md:86]: `focusFirstItem` isn't the _only_ effect this has hence the matching of the "keyboard" name

Comment (MarshallOfSound) [reply docs/api/menu.md:86]: > Also, do we ever foresee using any of the other [MenuSourceType](https://source.chromium.org/chromium/chromium/src/+/main:ui/base/ui_base_types.h;bpv=1;bpt=1;l=133?q=MENU_SOURCE_KEYBOARD&ss=chromium&gsn=MenuSourceType&gs=KYTHE%3A%2F%2Fkythe%3A%2F%2Fchromium.googlesource.com%2Fcodesearch%2Fchromium%2Fsrc%2F%2Fmain%3Flang%3Dc%252B%252B%3Fpath%3Dui%2Fbase%2Fui_base_types.h%23MenuSourceType%253Aui%2523n%2523TX9FKDk8TS) options available?

Lots of these are legacy / not used anymore so no not really

**Review by VerteDinde — APPROVED** [review]

File: lib/browser/api/menu.ts
Comment (clavin) [inline lib/browser/api/menu.ts:81]: Why default to `mouse`? Why not `none`?

**Review by clavin — APPROVED** [review]

**Review by zcbenz — APPROVED** [review]
I'm good with this, but we should probably wait for upstream's review on the patch before merging it.

File: docs/api/menu.md
Comment (samuelmaddock) [inline docs/api/menu.md:84]:
```suggestion
    provided by the `context-menu` event. It is not recommended to set this value manually,
```

**Review by samuelmaddock — APPROVED** [review]
API LGTM

File: lib/browser/api/menu.ts
Comment (MarshallOfSound) [reply lib/browser/api/menu.ts:81]: mouse was the old default, and is also the default upstream iirc.  Don't want to go breaking anything 😅

**Review by erickzhao — APPROVED** [review]
API LGTM

**Comment by MarshallOfSound** [comment]
The code change was +1'ed upstream so landing here https://chromium-review.googlesource.com/c/chromium/src/+/4656821

**Pull request merged**
